feat(core): support styles and styleUrl as strings

crisbeto opened the pull request:
Adds support for passing in `@Component.styles` as a string. Also introduces a new `styleUrl` property on `@Component` for providing a single stylesheet. This is more convenient for the most common case where a component only has one stylesheet associated with it.

eneajaho commented:
Hello @crisbeto 

This example below looks like it accepts multiple urls because of the naming. 
`styleUrls: 'dir/styles.css, dir/styles1.css'`

Imo, The correct way it should look like is: 
`styleUrl: 'dir/styles.css'`

But that would require big changes I believe.

crisbeto commented:
Yeah, I agree that the naming is problematic, but I think that splitting it on the comma would be weird as well. Introducing a `styleUrl` property won't be that much more work to support, but it increases the API surface. WDYT @jelbourn?

jnizet commented:
My two cents here: the Angular API already has at least one other place where it uses the plural form to also accept a single thing: https://angular.io/api/forms/AbstractControlOptions. So keeping `styleUrls` would at least be consistent.

crisbeto commented:
Based on what we discussed with @jelbourn: I've introduced a new `styleUrl` property and kept `styleUrls` as a `string[]`.

pkozlowski-opensource commented on packages/core/test/metadata/resource_loading_spec.ts:
```diff
@@ -122,6 +123,49 @@ Did you run and wait for 'resolveComponentResources()'?`.trim());
+      expect(resourceFetchCount).toBe(1);
+    });
+
+    it('should throw if both styles and styleUrl are passed in', async () => {
```
test name: `should throw if both styleUrls and styleUrl are passed in`

pkozlowski-opensource reviewed (approved):
LGTM

Reviewed-for: public-api 
Reviewed-for: fw-core

crisbeto committed:
feat(core): support styles and styleUrl as strings

Adds support for passing in `@Component.styles` as a string. Also introduces a new `styleUrl` property on `@Component` for providing a single stylesheet. This is more convenient for the most common case where a component only has one stylesheet associated with it.

jessicajaniuk reviewed (approved):
reviewed-for: fw-core, fw-testing, public-api

AndrewKushnir commented:
This PR was merged into the repository by commit 59387ee476dff1a893a01fe5cbee3c95b93c0cdb.

spock123 commented:
Thank you guys, I know this is a silly small thing but it's really something I bet all developers have come across: it really makes the developing experience better, because 99% of all cases there'll only be 1 styles file. Thanks!!!
